virtualbox/vbox-export-snapshots.py

```diff
@@ -32,56 +32,217 @@
+        raise Exception("Failed to create new hostonlyif.")
+    except Exception as e:
+        print(f"Error getting host-only interface name: {e}")
+    raise Exception("Failed to verify host-only interface exists")
+
+def change_network_adapters_to_hostonly(machine_guid):
+    """Changes all active network adapters to Host-Only. Must be poweredoff"""
+    ensure_hostonlyif_exists()
+    try:
+        foundOne = False
+        # change any existing enabled nic to hostonly
+        vminfo = run_vboxmanage(["showvminfo", machine_guid, "--machinereadable"])
+        for line in vminfo.splitlines():
+            # Match lines exactly in the format 'nicN="value"'
+            match = re.match(r"nic(\d+)=\"(.*?)\"", line)  
+            if match:
+                nic_number = match.group(1)
+                nic_value = match.group(2)
+                if nic_value != "none":  # Ignore NICs with value "none"
+                    run_vboxmanage(["modifyvm", machine_guid, f"--nic{nic_number}", "hostonly"])
+                    print(f"Changed nic{nic_number} to hostonly")
+                    foundOne = True
+        
+        # If no nic was enabled / configured, set the first to hostonly
+        if not foundOne:
+            run_vboxmanage(["modifyvm", machine_guid, f"--nic1", "hostonly"])
+
+        # ensure changes applied
+        vminfo = run_vboxmanage(["showvminfo", machine_guid, "--machinereadable"])
+        for line in vminfo.splitlines():
+            # Match lines exactly in the format 'nicN="value"'
+            match = re.match(r"nic(\d+)=\"(.*?)\"", line)  
+            if match:
+                nic_number = match.group(1)
+                nic_value = match.group(2)
+                if nic_value == "hostonly":
+                    print("Verified hostonly nic configuration correct")
+                    return
+    except Exception as e:
+        print(f"Error changing network adapters: {e}")
+    print("Failed to change VM network adapters to hostonly")
+    raise Exception("Failed to change VM network adapters to hostonly")
 
 if __name__ == "__main__":
     date = datetime.today().strftime("%Y%m%d")
 
-    vbox = virtualbox.VirtualBox()
-    vm = vbox.find_machine(VM_NAME)
-    max_adapters = vbox.system_properties.get_max_network_adapters(vm.chipset_type)
-
     for snapshot_name, extension, description in SNAPSHOTS:
+        print(f"Starting operations on {snapshot_name}")
         try:
-            # Restore snapshot
-            session = vm.create_session()
-            snapshot = session.machine.find_snapshot(snapshot_name)
-            progress = session.machine.restore_snapshot(snapshot)
-            progress.wait_for_completion(-1)
-            change_network_adapters(session.machine, max_adapters)
-            session.unlock_machine()
-            print(f"Restored '{snapshot_name}' and changed its adapter(s) to host-only")
-
+            vm_uuid = get_vm_uuid(VM_NAME)
```

Comment:
We could rely on the VM_NAME alone, but I use the UUID so that we can support multiple VMs of the same name and be sure we refer to the same VM consistently for all operations